HLTrigger/JetMET/plugins/HLTScoutingPFProducerV2.cc

```diff
@@ -0,0 +1,293 @@
+
+class HLTScoutingPFProducerV2 : public edm::global::EDProducer<> {
+public:
+  explicit HLTScoutingPFProducerV2(const edm::ParameterSet &);
+  ~HLTScoutingPFProducerV2() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
+
+private:
+  void produce(edm::StreamID sid, edm::Event &iEvent, edm::EventSetup const &setup) const final;
+
+  const edm::EDGetTokenT<reco::PFJetCollection> pfJetCollection_;
+  const edm::EDGetTokenT<reco::JetTagCollection> pfJetTagCollection_;
+  const edm::EDGetTokenT<reco::PFCandidateCollection> pfCandidateCollection_;
+  const edm::EDGetTokenT<reco::VertexCollection> vertexCollection_;
+  const edm::EDGetTokenT<reco::PFMETCollection> metCollection_;
+  const edm::EDGetTokenT<double> rho_;
+
+  const double pfJetPtCut_;
+  const double pfJetEtaCut_;
+  const double pfCandidatePtCut_;
+  const double pfCandidateEtaCut_;
+  const int mantissaPrecision_;
+
+  const bool doJetTags_;
+  const bool doCandidates_;
+  const bool doMet_;
+  const bool doCandIndsForJets_;
+};
+
+//
+// constructors and destructor
+//
+HLTScoutingPFProducerV2::HLTScoutingPFProducerV2(const edm::ParameterSet &iConfig)
+    : pfJetCollection_(consumes(iConfig.getParameter<edm::InputTag>("pfJetCollection"))),
+      pfJetTagCollection_(consumes(iConfig.getParameter<edm::InputTag>("pfJetTagCollection"))),
+      pfCandidateCollection_(consumes(iConfig.getParameter<edm::InputTag>("pfCandidateCollection"))),
+      vertexCollection_(consumes(iConfig.getParameter<edm::InputTag>("vertexCollection"))),
+      metCollection_(consumes(iConfig.getParameter<edm::InputTag>("metCollection"))),
+      rho_(consumes(iConfig.getParameter<edm::InputTag>("rho"))),
+      pfJetPtCut_(iConfig.getParameter<double>("pfJetPtCut")),
+      pfJetEtaCut_(iConfig.getParameter<double>("pfJetEtaCut")),
+      pfCandidatePtCut_(iConfig.getParameter<double>("pfCandidatePtCut")),
+      pfCandidateEtaCut_(iConfig.getParameter<double>("pfCandidateEtaCut")),
+      mantissaPrecision_(iConfig.getParameter<int>("mantissaPrecision")),
+      doJetTags_(iConfig.getParameter<bool>("doJetTags")),
+      doCandidates_(iConfig.getParameter<bool>("doCandidates")),
+      doMet_(iConfig.getParameter<bool>("doMet")),
+      doCandIndsForJets_(iConfig.getParameter<bool>("doCandIndsForJets")) {
+  //register products
+  produces<Run3ScoutingPFJetCollection>();
+  produces<Run3ScoutingParticleV2Collection>();
+  produces<double>("rho");
+  produces<double>("pfMetPt");
+  produces<double>("pfMetPhi");
+}
+
+HLTScoutingPFProducerV2::~HLTScoutingPFProducerV2() = default;
+
+// ------------ method called to produce the data  ------------
+void HLTScoutingPFProducerV2::produce(edm::StreamID sid, edm::Event &iEvent, edm::EventSetup const &setup) const {
+  using namespace edm;
+
+  //get vertices
+  Handle<reco::VertexCollection> vertexCollection;
+  std::unique_ptr<Run3ScoutingVertexCollection> outVertices(new Run3ScoutingVertexCollection());
```

Comment:
`std::make_unique` is preferred over `new`, e.g.
```suggestion
  auto outVertices = std::make_unique<Run3ScoutingVertexCollection>();
```
(also elsewhere)

Reply:
Fixed it for all applicable cases in [this commit](https://github.com/cms-sw/cmssw/pull/36840/commits/f933d59922918b5c0c43b53e01f30c1c9233702c#).